tests/invoke_test.go

```diff
@@ -66,57 +54,81 @@ func Test_Invoke_With_Supported_Verbs(t *testing.T) {
-		FunctionName: "env-test",
-		Network:      "func_functions",
-		FProcess:     "env",
-		EnvVars:      envVars,
-	}
-
-	deployStatus := deploy(t, functionRequest)
-	if deployStatus != http.StatusOK && deployStatus != http.StatusAccepted {
-		t.Fatalf("got %d, wanted %d or %d", deployStatus, http.StatusOK, http.StatusAccepted)
-	}
-
-	list(t, http.StatusOK)
-
+func invokeWithCustomEnvVarsAndQueryString(t *testing.T, functionRequest *sdk.DeployFunctionSpec) {
 	t.Run("Empty QueryString", func(t *testing.T) {
-		bytesOut := invoke(t, functionRequest.FunctionName, emptyQueryString, "", http.StatusOK)
+		bytesOut := invoke(t, functionRequest, emptyQueryString, "", http.StatusOK)
 		out := string(bytesOut)
 		if strings.Contains(out, "custom_env") == false {
 			t.Fatalf("want: %s, got: %s", "custom_env", out)
 		}
 	})
 
 	t.Run("Populated QueryString", func(t *testing.T) {
-		bytesOut := invoke(t, functionRequest.FunctionName, "testing=1", "", http.StatusOK)
+		bytesOut := invoke(t, functionRequest, "testing=1", "", http.StatusOK)
 		out := string(bytesOut)
 		if strings.Contains(out, "Http_Query=testing=1") == false {
 			t.Fatalf("want: %s, got: %s", "Http_Query=testing=1", out)
 		}
 	})
 }
+
+func Test_Invoke(t *testing.T) {
+	cases := []FunctionTestCase{
+		{
+			name: "Invoke test with different verbs",
+			function: types.FunctionDeployment{
+				Image:      "functions/alpine:latest",
+				Service:    "env-test-verbs",
+				EnvProcess: "env",
+				EnvVars:    map[string]string{},
+				Namespace:  config.DefaultNamespace,
+			},
+		},
+		{
+			name: "Invoke propogates redirect to the caller",
+			function: types.FunctionDeployment{
+				Image:      "theaxer/redirector:latest",
+				Service:    "redirector-test",
+				EnvProcess: "./handler",
+				EnvVars:    map[string]string{"destination": "http://example.com"},
+				Namespace:  config.DefaultNamespace,
+			},
+		},
+		{
+			name: "Invoke with custom env vars and query string",
+			function: types.FunctionDeployment{
+				Image:      "functions/alpine:latest",
+				Service:    "env-test",
+				EnvProcess: "env",
+				EnvVars:    map[string]string{"custom_env": "custom_env_value"},
+				Namespace:  config.DefaultNamespace,
+			},
+		},
+	}
+
+	cases = copyNamespacesTest(cases)
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			functionRequest := createDeploymentSpec(c)
+			deployStatus := deploy(t, functionRequest)
+			if deployStatus != http.StatusOK && deployStatus != http.StatusAccepted {
+				t.Fatalf("got %d, wanted %d or %d", deployStatus, http.StatusOK, http.StatusAccepted)
+				return
+			}
+
+			list(t, http.StatusOK, functionRequest.Namespace)
+
+			switch service := c.function.Service; service {
+			case "env-test-verbs":
+				invokeWithSupportedVerbs(t, functionRequest)
+			case "redirector-test":
+				_ = invoke(t, functionRequest, emptyQueryString, "", http.StatusFound)
+			case "env-test":
+				invokeWithCustomEnvVarsAndQueryString(t, functionRequest)
```

Comment:
i feel like this switch based on function names is a bit brittle and will make this harder to extend in the future. But I think we can keep it for now and address it later.